ArduPlane/fence.cpp

```diff
@@ -0,0 +1,132 @@
+#include "Plane.h"
+
+// Code to integrate AC_Fence library with main ArduPlane code
+
+#if AC_FENCE == ENABLED
+
+// fence_check - ask fence library to check for breaches and initiate the response
+// called at 1Hz
+void Plane::fence_check()
+{
+    const uint8_t orig_breaches = fence.get_breaches();
+
+    // check for new breaches; new_breaches is bitmask of fence types breached
+    const uint8_t new_breaches = fence.check();
+
+    // we still don't do anything when disarmed, but we do check for fence breaches.
+    // fence pre-arm check actually checks if any fence has been breached
+    // that's not ever going to be true if we don't call check on AP_Fence while disarmed
+    if (!arming.is_armed())
+    {
+        return;
+    }
+
+    if (!fence.enabled())
+    {
```

Comment:
Brace on preceding line; similarly elsewhere.

Reply:
On a dev call it was mentioned there was a script that performs these checks? My google-fu is weak today, apparently.